### ISSUE TITLE
Which version of incanter works with older JRE's

### ISSUE THREAD

**Issue by mschuene** [incanter/incanter#344]
I have 
[incanter/incanter-core "1.5.7"]
[incanter/incanter-charts "1.5.7"]

in my project dependencies.

When the program is run on jre7 I get the error message
Incanter/Matrix: unsupported major.minor version 52.0

It seems that part of incanter (Incanter/Matrix for example) is compiled against java 8 and that this causes the error during class loading on java 7 or lower.

Do you know which versions of incanter require which jre version? Can the current version be made compatible with jre7 or jre6?

**Comment by kittylyst** [comment]
The question is whether that is desirable.

Java 6 is *ancient*. It was superceded 5 years ago, and has been EOL'd with no further public releases since 3 years ago. The bytecode format is substantially different from 7 & 8 with no good way to backport certain features. It is also incredibly slow compared with 7 or 8 and has horrendous security risks (dozens of known attacks that will never be fixed). So why should any modern software project hamstring itself by supporting such a dinosaur? - the correct response is to require Java 6 sites to upgrade.

Java 7 is in (a bit) better shape, but was still EOL'd over 12 months ago. There are still significant differences in the bytecode format, mostly around lambdas & how they're supported (as well as the class library support of course). Perhaps more of an argument could be made for jre7, but it's still a considerable amount of work for a minority (& declining) platform, that also could inhibit future development by disallowing modern Java / JVM features or requiring additional complexity, shims, etc.

Bottom line: jre8 is the fastest adopted Java version in history, is fast & secure and given the changes coming in Java 9, jre8 looks set to be with us for a very long time. My usual advice is to consider 8 as a "long-term support" release, and spend effort getting applications upgraded to that version and plan to stay there for the forseeable future.

**Comment by mikera** [comment]
I would definitely suggest dropping support for Java 6. As @kittylyst says it is truly ancient and pretty much a waste of time to support nowadays (unless someone with deep pockets is willing to pay you to do it...). Java 7 I'd personally be happy to drop, but I know quite a few are still using it.

A slightly bigger issue is the fact that we have a version-specific compilation. In my view, there is nothing in Incanter that should really require this.

Note that 1.5.7 is the legacy branch of Incanter... can you not switch to 1.9.1 now?

**Comment by mschuene** [comment]
@mikera  1.5.7 is still the version on the readme so I just picked that :) But 1.9.1 also doesn't work with jre7.

I wanted to use incanter simply to draw a graph and the application should also run under jre7 -> I probably will use jfreecharts directly for that

Nevertheless I think it would be great not to have version-specific compilation if it is not too much effort.

**Comment by mikera** [comment]
I think I may have fixed in this PR: https://github.com/incanter/incanter/pull/345

Can you test? If that has solved the problem I will make a release

**Comment by mschuene** [comment]
@mikera This fixed it for me, thanks!
